### ISSUE TITLE
[Breaking] Trainer completely breaks on GCP TPU VMs

### ISSUE THREAD

**Issue by siddk** [Lightning-AI/lightning#13088]
## 🐛 Bug

Running the Lightning Trainer with TPUs on GCP TPU VMs leads to complete training failures - processing a single batch with any model takes forever, and eventually (on the order of ~20 batches, or 3 minutes!) **completely stalls**. This is notably _with the provided MNIST example_.

However, writing the equivalent "native" PyTorch code has no problems, completing a full epoch (~54 batches of MNIST) in only 8 seconds. Full code, logs (stdout & stderr), and XLA profiler report are in [this Gist](https://gist.github.com/siddk/bbdc3692de1fad66c4d4dc621a6c2bcf), with components listed below.

This is a semi-critical bug that's seriously affecting adopting PyTorch-Lightning for TPU workloads! The "native" loop only takes 8 seconds per epoch, but using the Trainer stalls at 20 batches in 2m30s. 

### Environment

This is using a TPU-v2-8 VM (but only training on a Single Core) with the default PyTorch 1.11 VM. I've checked versions for all necessary libraries, and include them below.

- PyTorch Lightning Version (e.g., 1.5.0): 1.6.3 (latest, stable)
- PyTorch Version (e.g., 1.10): 1.11
- PyTorch XLA: 1.11
- Python version (e.g., 3.9): 3.8.10
- OS (e.g., Linux): Linux
- Any other relevant information: This is using the new "one-stop" TPU VMs, not the older (soon-to-be deprecated) TPU Nodes.

### To Reproduce

- "Native" PyTorch Script with handwritten training loop and TPU device placement: [`debug_native.py`](https://gist.github.com/siddk/bbdc3692de1fad66c4d4dc621a6c2bcf#file-debug_native-py)
    + Produces [this log file w/ stdout & stderr](https://gist.github.com/siddk/bbdc3692de1fad66c4d4dc621a6c2bcf#file-debug-native-log)
    + Also dumped [output of the XLA metrics report](https://gist.github.com/siddk/bbdc3692de1fad66c4d4dc621a6c2bcf#file-native-xla-report-txt)

- Lightning Trainer Script: [`debug_lightning.py`](https://gist.github.com/siddk/bbdc3692de1fad66c4d4dc621a6c2bcf#file-debug_lightning-py)
    + **Important:** Training stalled out ~20 batches... had to kill process, impacting logging.
    + Produces [this log file w/ stdout & stderr](https://gist.github.com/siddk/bbdc3692de1fad66c4d4dc621a6c2bcf#file-debug-lightning-log)
    + Also dumped [output of the XLA metrics report](https://gist.github.com/siddk/bbdc3692de1fad66c4d4dc621a6c2bcf#file-lightning-xla-report-txt)

Both python scripts are standalone, and should be runnable on TPU devices as-is.

### Additional context

Not immediately clear what the problem is from the XLA metrics, but one thing to note is that the default behavior of the Lightning Trainer is to print the train loss in the progress_bar every step... this call hides an implicit `.item()` which is bad for TPUs/XLA?

Some runs with the Lightning Trainer (with `PT_XLA_DEBUG=1`) also sometimes point to `pytorch_lightning.utilities.data` and the function `has_len_all_ranks()` as implementing an "unlowered op"... but this alone doesn't account for the huge performance discrepancy.

cc @kaushikb11 @rohitgr7 @borda @akihironitta

**Comment by carmocca** [comment]
We recently made some TPU fixes (cc @awaelchli). Would you mind trying on the latest version? (1.7.0rc0)

**Comment by adamcatto** [comment]
I'm getting similar behavior on pt-lightning 1.7.0 with all same specs except running on GCP TPU-v3-8 VM. 70 seconds per batch, breaks after 13 batches. (for reference: batch size of 128 images of dimension 96x96 with resent-50 backbone, tried 4096 previously but it just breaks with out-of-HBM error. For the record, I could perform distributed training on 4 V100 GPUS with batch size of 4096 with the exact same code; seems to be TPU-specific. I had also written my code with native PyTorch-XLA on Colab TPU and it worked fine, seems that translating to pytorch-lightning and running on GCP broke it?)

**Comment by carmocca** [comment]
We now run tests on gcloud and don't see any issues related to the top post. Closing

Happy to discuss if this is still a problem
